resources/top_level_overview.markdown

```diff
@@ -0,0 +1,283 @@
+writes sent back to `master` for use by the reduce workers.
+
+5. When the `master` notifies a _reduce_ worker about a particular location of
+_intermediate_ key-value pairs, it reads the buffered data from the local
+disks of the map workers. When a reduce worker has read all intermediate data,
+it sorts it by key so that all of the same keys are grouped together. The
+sorting is needed because many different keys can map to the same reduce task.
+If the amount of intermediate data is too large to fit in memory, an external
+sort is used.
+
+6. The reduce worker iterates over the sorted data and for each unique
+intermediate key encountered, it passes the key and the corresponding set of
+intermediate values to the user’s _reduce_ function. The output of the reduce
+function is appended to a final output file for this reduce partition.
+
+7. When all map tasks and reduce tasks have been completed, the call to
+MapReduce reaches a return statement back to user code.
+
+## How Hadoop Handles MapReduce
+Now that we are comfortable with what MapReduce is, let's look into
+understanding how Hadoop handles MapReduce.  Here, each point expands on the
+corresponding number in [Steps in a MapReduce Process](#steps-in-a-map-reduce-process).
+
+Before a job can be started, however, it must run through a few steps:
+1. A job must be submitted via the user API.
+2. The resource manager retrieves the job ID (application ID)
+3. The job client checks output specification, computes splits (which can also
+  be changed to be computed over the cluster, useful for a job with many splits),
+  and then copy job resources over to the HDFS.
+
+#### Point 1 & 2: Splitting and Assigning Tasks
+HDFS (Hadoop Distributed File System) breaks down large files into configurable
+sized blocks, defaulting to 64 MB.  It then stores copies of these blocks across
+nodes in the cluster.  
+
+Using YARN (YARN Applied Resource Negotiator), The resource manager acts as the
+cluster resource manager (no way), as well as the job scheduling facility.  The
+resource manager creates an Application Master daemon (AMD) to monitor the
+life cycle of the job.  One of the initial things that the AMD does is distinguish
+which HDFS blocks are needed for processing.  The AMD requests data from the
+`NameNode` on where the relevant replicated blocks of data are stored. With this
+information, the AMD then requests the resource manager to have map tasks
+process the specified blocks on the same nodes where the data is stored. This
+data locality helps to avoid potentially painful network bottlenecks.
+
+The number of map tasks correspond to the number of created blocks, while the
+number of reducers are configurable via the `mapreduce.jobs.reduces` property.
+Thus for each input split, a map task is spawned.  The number of maps is
+typically driven by the number of HDFS blocks in the input file. This block size
+is configurable, and therefore allows for a workaround when wishing to adjust the
```

Comment:
The sentence "The number of maps..." seems redundant, and I think I would remove it